components/camel-opensearch/src/test/java/org/apache/camel/component/opensearch/integration/OpensearchGetSearchDeleteExistsUpdateIT.java

```diff
@@ -0,0 +1,913 @@
+
+        //now, verify GET succeeded
+        GetResponse<?> response = template.requestBody("direct:get", indexId, GetResponse.class);
+        assertNotNull(response, "response should not be null");
+        assertNotNull(response.source(), "response source should not be null");
+
+        //now, perform Delete
+        Result deleteResponse = template.requestBody("direct:delete", indexId, Result.class);
+        assertNotNull(deleteResponse, "response should not be null");
+
+        //now, verify GET fails to find the indexed value
+        response = template.requestBody("direct:get", indexId, GetResponse.class);
+        assertNotNull(response, "response should not be null");
+        assertNull(response.source(), "response source should be null");
+    }
+
+    @Test
+    void testSearchWithMapQuery() {
+        //first, Index a value
+        Map<String, String> map1 = Map.of("testSearchWithMapQuery1", "foo");
+        Map<String, String> map2 = Map.of("testSearchWithMapQuery2", "bar");
+        Map<String, Object> headers = Map.of(
+                OpensearchConstants.PARAM_OPERATION, OpensearchOperation.Bulk,
+                OpensearchConstants.PARAM_INDEX_NAME, "twitter");
+        template.requestBodyAndHeaders("direct:start", List.of(Map.of("doc", map1), Map.of("doc", map2)), headers,
+                String.class);
+
+        // No match
+        Map<String, Object> actualQuery = new HashMap<>();
+        actualQuery.put("doc.testSearchWithMapQuery1", "bar");
+        Map<String, Object> match = new HashMap<>();
+        match.put("match", actualQuery);
+        Map<String, Object> query = new HashMap<>();
+        query.put("query", match);
+        HitsMetadata<?> response = template.requestBody("direct:search", query, HitsMetadata.class);
+        assertNotNull(response, "response should not be null");
+        assertNotNull(response.total());
+        assertEquals(0, response.total().value(), "response hits should be == 0");
+
+        // Match
+        actualQuery.put("doc.testSearchWithMapQuery1", "foo");
+        // the result may see stale data so use Awaitility
+        Awaitility.await().atMost(10, TimeUnit.SECONDS).untilAsserted(() -> {
+            HitsMetadata<?> resp = template.requestBody("direct:search", query, HitsMetadata.class);
+            assertNotNull(resp, "response should not be null");
+            assertNotNull(resp.total());
+            assertEquals(1, resp.total().value(), "response hits should be == 1");
+            assertEquals(1, resp.hits().size(), "response hits should be == 1");
+            Object result = resp.hits().get(0).source();
+            assertInstanceOf(ObjectNode.class, result);
+            assertTrue(((ObjectNode) result).has("doc"));
+            JsonNode node = ((ObjectNode) result).get("doc");
+            assertTrue(node.has("testSearchWithMapQuery1"));
+            assertEquals("foo", node.get("testSearchWithMapQuery1").asText());
+        });
```

Comment:
I think it would be better to split this so that `untilAsserted` tests for only 1 assertion. Combining them usually makes it harder to identify the root cause of failures.